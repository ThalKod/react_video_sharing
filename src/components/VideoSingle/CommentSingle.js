import React from "react";
import { Link } from "react-router-dom"

import Avatar from "components/Header/Avatar";
import { formatTimestamps } from "utils";


export default class CommentSingle extends React.Component{

  state= {
    replyBox: false
  };

  reply = () => {
    this.setState({ replyBox: true});
  };

  render(){
    const { text, author: { username, _id }, createdAt } = this.props;
    const { replyBox } = this.state;

    return(
        <div className="cl-comment">
          <div className="cl-avatar">
            <Link to={`/channel/${_id}`}>
              <Avatar username={username}/>
            </Link>
          </div>
          <div className="cl-comment-text">
            <div className="cl-name-date"><Link to={`/channel/${_id}`}>{username}</Link> . {formatTimestamps(createdAt)}</div>
            <div className="cl-text">{text}</div>
            { !replyBox ?
                <div className="cl-meta">
                  <button onClick={this.reply} className="reply-button" type="submit">Reply</button>
                </div>
                :
                <div className="reply-comment">
                  <div className="rc-comment">
                    <form>
                      <textarea
                          rows="3"
                          placeholder="Enter comment here..."
                      />
                      <button type="submit" onClick={this.handleCommentSubmit}>
                        <i className="cv cvicon-cv-add-comment"/>
                      </button>
                    </form>
                  </div>
                </div>
            }
          </div>
          <div className="clearfix"/>
        </div>
    )
  }
}

