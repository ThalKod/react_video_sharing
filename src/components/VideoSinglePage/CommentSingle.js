import React from "react";
import { Link } from "react-router-dom"

import Avatar from "components/Header/Avatar";
import { formatTimestamps } from "utils";

export default function CommentSingle({ text, author: { username, _id }, createdAt }){
  return(
      <div className="cl-comment">
        <div className="cl-avatar">
          <Link to={`/channel/${_id}`}>
            <Avatar username={username}/>
          </Link>
        </div>
        <div className="cl-comment-text">
          <div className="cl-name-date"><Link to={`/channel/${_id}`}>{username}</Link> . {formatTimestamps(createdAt)}</div>
          <div className="cl-text">{text}</div>
        </div>
        <div className="clearfix"/>
      </div>
  )
}
