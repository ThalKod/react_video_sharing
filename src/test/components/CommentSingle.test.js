import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json"

import CommentSingle from "../../components/VideoSingle/CommentSingle";

describe("CommentSingle", () => {
  it("should render correctly the CommentSingle components", () => {
    const props = {
      text: "Hello World !",
      author: { username: "My name" },
      createdAt: 1129910012010,
      reply: []
    };
    const wrapper = shallow(<CommentSingle {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
