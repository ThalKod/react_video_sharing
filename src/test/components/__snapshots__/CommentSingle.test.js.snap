// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentSingle should render correctly the CommentSingle components 1`] = `
<div
  className="cl-comment"
>
  <div
    className="cl-avatar"
  >
    <Link
      replace={false}
      to="/"
    >
      <_default
        username="My name"
      />
    </Link>
  </div>
  <div
    className="cl-comment-text"
  >
    <div
      className="cl-name-date"
    >
      <a
        href="/"
      >
        My name
      </a>
       . 
      13 years ago
    </div>
    <div
      className="cl-text"
    >
      Hello World !
    </div>
  </div>
  <div
    className="clearfix"
  />
</div>
`;
